public/js/reducers/sources.js

```diff
@@ -81,13 +79,19 @@ function update(state = initialState, action) {
   return state;
 }
 
-function getFilenameFromUrl(sourceUrl) {
+function _updateSource(source) {
+  const pathname = getPathnameFromUrl(source.url);
+  const filename = pathname.substring(pathname.lastIndexOf("/") + 1);
+  return Object.assign({}, source, { pathname, filename });
+}
+
+function getPathnameFromUrl(sourceUrl) {
   if (!sourceUrl) {
     return "";
   }
 
   const url = new URL(sourceUrl);
```

Comment:
> We don't test components in node, we'll run them in the browser

Yea, we can decide to do that. Lets wait and see though, it might be simpler or significantly faster to run in node. Not sure yet.

> selectors

I'll move it over now